packages/paste-core/components/badge/src/types.ts

```diff
@@ -0,0 +1,11 @@
+import type {MouseEventHandler, ReactNode} from 'react';
+
+export type BadgeVariants = 'info' | 'default' | 'warning' | 'error' | 'success' | 'new';
+export type BadgeBaseElements = 'button' | 'span' | 'a';
+export interface BadgeProps {
+  variant: BadgeVariants;
+  children: NonNullable<ReactNode>;
+  as?: BadgeBaseElements;
```

Comment:
didn't we decide that as was a required prop?

Reply:
I thought it was that in order for the badge to be a button or anchor that both `as` and `onClick`/`href` (respectively) are required, the same as it is in button. We def decided that `variant` is required.

Reply:
I still think this should be required.